client/asset/eth/eth.go

```diff
@@ -557,11 +647,14 @@ func (eth *ExchangeWallet) FundOrder(ord *asset.Order) (asset.Coins, []dex.Bytes
 			ord.DEXConfig.MaxFeeRate,
 			eth.gasFeeLimit)
 	}
+
 	maxSwapFees := ord.DEXConfig.MaxFeeRate * ord.DEXConfig.SwapSize * ord.MaxSwapCount
-	refundFees := dexeth.RefundGas(0) * eth.gasFeeLimit
-	fundsNeeded := ord.Value + maxSwapFees + refundFees
-	coins := asset.Coins{eth.createFundingCoin(fundsNeeded)}
-	err := eth.lockFunds(fundsNeeded)
+	initiationFunds := ord.Value + maxSwapFees
+	coins := asset.Coins{eth.createFundingCoin(initiationFunds)}
+
+	eth.lockedFundsMtx.Lock()
+	defer eth.lockedFundsMtx.Unlock()
+	err := eth.lockedFunds.lock(initiationFunds, initiationReserve, eth.balance)
```

Comment:
I was thinking about that too, but I did it this way because if the refund reserves were locked here, then `FundOrder` would need to return the amount locked for refunds separately. I didn't want this to have to be done in non-account based coins.

Reply:
I see what you mean. I might propose a change later, but this is good for now.